File: app/src/main/java/com/mindbox/app/Certificate.java
package com.mindbox.app;

import java.util.UUID;

public class Certificate {

    private String id;
    private String title;
    private String platform;
    private String issueDate;
    private String date; // Fecha de registro
    private String folio;
    private String credlyId;
    private String score;
    private String notes;
    private String pdfUrl;
    private long timestamp;

    // Constructor requerido por Firebase
    public Certificate() {
        this.id = UUID.randomUUID().toString();
        this.platform = "Otro";
    }

    // Getters
    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getPlatform() {
        return platform;
    }

    public String getIssueDate() {
        return issueDate;
    }

    public String getDate() {
        return date;
    }

    public String getFolio() {
        return folio;
    }

    public String getCredlyId() {
        return credlyId;
    }

    public String getScore() {
        return score;
    }

    public String getNotes() {
        return notes;
    }

    public String getPdfUrl() {
        return pdfUrl;
    }

    public long getTimestamp() {
        return timestamp;
    }

    // Setters
    public void setId(String id) {
        this.id = id;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public void setIssueDate(String issueDate) {
        this.issueDate = issueDate;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setFolio(String folio) {
        this.folio = folio;
    }

    public void setCredlyId(String credlyId) {
        this.credlyId = credlyId;
    }

    public void setScore(String score) {
        this.score = score;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public void setPdfUrl(String pdfUrl) {
        this.pdfUrl = pdfUrl;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: lib/data/models/certificate.dart
import 'package:uuid/uuid.dart';

class Certificate {
  final String id;
  final String title;
  final String platform;
  final String date;
  final String issueDate;
  final String? folio;
  final String? credlyId;
  final String? credlyIssuer;
  final String? score;
  final String? notes;
  final String? pdfUrl;

  Certificate({
    String? id,
    required this.title,
    this.platform = 'Otro',
    this.date = '',
    this.issueDate = '',
    this.folio,
    this.credlyId,
    this.credlyIssuer,
    this.score,
    this.notes,
    this.pdfUrl,
  }) : id = id ?? const Uuid().v4();

  factory Certificate.fromJson(Map<String, dynamic> json) {
    return Certificate(
      id: json['\$id'] ?? json['id'],
      title: json['title'] ?? '',
      platform: json['platform'] ?? 'Otro',
      date: json['date'] ?? '',
      issueDate: json['issueDate'] ?? '',
      folio: json['folio'],
      credlyId: json['credlyId'],
      credlyIssuer: json['credlyIssuer'],
      score: json['score'],
      notes: json['notes'],
      pdfUrl: json['pdfUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'platform': platform,
      'date': date,
      'issueDate': issueDate,
      'folio': folio,
      'credlyId': credlyId,
      'credlyIssuer': credlyIssuer,
      'score': score,
      'notes': notes,
      'pdfUrl': pdfUrl,
    };
  }

  Certificate copyWith({
    String? id,
    String? title,
    String? platform,
    String? date,
    String? issueDate,
    String? folio,
    String? credlyId,
    String? credlyIssuer,
    String? score,
    String? notes,
    String? pdfUrl,
  }) {
    return Certificate(
      id: id ?? this.id,
      title: title ?? this.title,
      platform: platform ?? this.platform,
      date: date ?? this.date,
      issueDate: issueDate ?? this.issueDate,
      folio: folio ?? this.folio,
      credlyId: credlyId ?? this.credlyId,
      credlyIssuer: credlyIssuer ?? this.credlyIssuer,
      score: score ?? this.score,
      notes: notes ?? this.notes,
      pdfUrl: pdfUrl ?? this.pdfUrl,
    );
  }
}
